matchbox_socket/src/webrtc_socket/native/message_loop.rs

```diff
@@ -483,30 +518,43 @@ async fn peer_loop(
         Output = Result<
             (
                 PeerId,
-                Arc<RTCDataChannel>,
+                Vec<Arc<RTCDataChannel>>,
                 Pin<Box<dyn FusedFuture<Output = Result<(), Box<dyn std::error::Error>>> + Send>>,
             ),
             Box<dyn std::error::Error>,
         >,
     >,
-    mut to_peer_message_rx: UnboundedReceiver<Packet>,
+    mut to_peer_message_rx: Vec<UnboundedReceiver<Packet>>,
 ) {
-    let (_peer_id, data_channel, mut trickle_fut) = handshake_fut.await.unwrap();
-
-    let message_loop_fut = async move {
-        while let Some(message) = to_peer_message_rx.next().await {
-            debug!("tx {:?}", message);
-            let message = message.clone();
-            let message = Bytes::from(message);
-            data_channel.send(&message).await.unwrap();
-        }
-    };
-    let message_loop_fut = message_loop_fut.fuse();
-    pin_mut!(message_loop_fut);
+    let (_peer_id, mut data_channels, mut trickle_fut) = handshake_fut.await.unwrap();
+
+    let mut futures = FuturesUnordered::new();
+
+    assert_eq!(
+        data_channels.len(),
+        to_peer_message_rx.len(),
+        "amount of data channels and receivers differ"
+    );
+    for _ in 0..data_channels.len() {
```

Comment:
could this be rewritten using `.zip`, `.map` + `.collect` instead of `pop` and `push` and for?

Reply:
Yes that's better